src-tauri/src/bundler.rs

```diff
@@ -337,3 +337,84 @@ impl Hook for NoopHook {
         unimplemented!();
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use parameterized::parameterized;
+    use pretty_assertions::assert_eq;
+    use std::fs::read_to_string;
+
+    /// Assert that an [`Error`] object has the expected chain of reasons.
+    fn assert_err_eq(error: Error, chain: Vec<String>) {
+        let mut error_chain = error.chain();
+        for expected_msg in chain {
+            assert_eq!(
+                error_chain.next().map(|msg| format!("{msg}")),
+                Some(expected_msg)
+            );
+        }
+        // Assert that the chain of reasons ends here
+        assert_eq!(error_chain.next().map(|msg| format!("{msg}")), None);
+    }
+
+    #[parameterized(case = {
+        "no_react",  // Not explicitly using React
+        "with_react_hook",  // Using React hook
+        "import_relative",  // Relative import with extension
+        "import_relative_no_ext",
+        "import_relative_directory",
+        "import_jsx",
+        "import_no_inner_React",
+        "import_no_outer_React"
+    })]
+    fn test_bundle_basic(case: &str) {
+        // Test the most basic bundler functionalities, with no dependencies and are
+        // expected to succeed
+        let root = PathBuf::from(format!("tests/fixtures/bundler/{case}/input"))
+            .canonicalize()
+            .unwrap();
+        let result = bundle(&root, root.join("index.jsx").as_path(), None)
+            .expect("Failed to bundle");
+
+        let expected =
+            read_to_string(format!("tests/fixtures/bundler/{case}/output.js")).unwrap();
+        self::assert_eq!(result, expected);
+    }
+
+    #[test]
+    fn test_bundle_absolute_import_error() {
+        // Test that we do not allow absolute path import
+        let root_path = "tests/fixtures/bundler/import_absolute/input";
+        let root = PathBuf::from(root_path);
+        let error = bundle(&root, root.join("index.jsx").as_path(), None)
+            .expect_err("Expected an error");
+
+        let expected = vec![
+            "load_transformed failed".to_string(),
+            "failed to analyze module".to_string(),
+            format!("failed to resolve /usr/bin/script.js from {root_path}/index.jsx"),
+            "Absolute imports are not supported; use relative imports instead"
+                .to_string(),
+        ];
+        assert_err_eq(error, expected);
+    }
+
+    #[test]
+    fn test_bundle_beyond_root_error() {
+        // Test that we do not allow absolute path import
+        let root_path = "tests/fixtures/bundler/import_beyond_root/input";
+        let root = PathBuf::from(root_path);
+        let error = bundle(&root, root.join("index.jsx").as_path(), None)
+            .expect_err("Expected an error");
+        let abs_root = root.canonicalize().expect("Failed to get absolute path");
```

Comment:
```suggestion
        let abs_root = root.canonicalize().unwrap();
```

I would suggest using `unwrap` here since it is not directly related to the function we test.